grafana-oncall.yaml

```diff
@@ -0,0 +1,196 @@
+      - py${{vars.py-version}}-setuptools
+      - py${{vars.py-version}}-tz
+      - python-${{vars.py-version}}
+      - python-${{vars.py-version}}-dev
+      - tzdata
+      - uv
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/grafana/oncall
+      tag: v${{package.version}}
+      expected-commit: 39f1d3baedbf8dc8ec0f92babfecb2362fcd38c0
+
+  - working-directory: engine
+    pipeline:
+      - name: Create venv
+        runs: |
+          # We use uv here as pip backtracks in resolution
+          uv venv -p python${{vars.py-version}} --seed
+      - name: Install grpcio (aarch64)
+        if: ${{build.arch}} == "aarch64" # Upstream only install grpcio on aaarch64
+        runs: |
+          # git lfs install
+          # git lfs pull
+          # uv pip install --no-compile grpcio-*.whl
+          rm -rf grpcio-*.whl
+      - name: Install requirements
+        runs: |
+          uv pip install --no-cache -r requirements.txt --no-compile
+      - name: Mitigate CVEs
+        runs: |
+          # Mitigates CVE-2024-56374, CVE-2024-56201 and CVE-2024-56326
+          uv pip install --no-cache --no-compile --upgrade django==4.2.18 jinja2==3.1.5
+      - name: Move venv
+        runs: |
+          mkdir -p ${{targets.destdir}}/usr/share/grafana-oncall/
+          mv .venv ${{targets.destdir}}/usr/share/grafana-oncall/
+      - name: Copy source
+        runs: |
+          cp -aR . ${{targets.destdir}}/usr/share/grafana-oncall/
+      - name: Create database
+        runs: |
+          mkdir -p ${{targets.destdir}}/var/lib/oncall
+          DJANGO_SETTINGS_MODULE=settings.prod_without_db DATABASE_TYPE=sqlite3 DATABASE_NAME=${{targets.destdir}}/var/lib/oncall/oncall.db SECRET_KEY="ThEmUsTSecretKEYforBUILDstage123" SILK_PROFILER_ENABLED="True" ${{targets.destdir}}/usr/share/grafana-oncall/.venv/bin/python manage.py collectstatic --no-input
+          DATABASE_TYPE=sqlite3 DATABASE_NAME=${{targets.destdir}}/var/lib/oncall/oncall.db ${{targets.destdir}}/usr/share/grafana-oncall/.venv/bin/python manage.py create_sqlite_db
+          stat ${{targets.destdir}}/var/lib/oncall/oncall.db # Sanity check
+      - name: Creeate required directories
+        runs: |
+          mkdir -p ${{targets.destdir}}/tmp/silk_profiles
+          mkdir -p ${{targets.destdir}}/tmp/prometheus_django_metrics
+      - name: Update venv paths
+        runs: |
+          # Replace full build path with target path
+          find ${{targets.destdir}}/usr/share/grafana-oncall/.venv/bin/ -type f | xargs sed -i "s|/home/build/engine|/usr/share/grafana-oncall|g"
+          # Existing configuration updates
+          sed -i "s|include-system-site-packages = false|include-system-site-packages = true|g" ${{targets.destdir}}/usr/share/grafana-oncall/.venv/pyvenv.cfg
+          sed -i "s|/home/build.*python${{vars.py-version}}|/usr/bin/python${{vars.py-version}}|" ${{targets.destdir}}/usr/share/grafana-oncall/.venv/pyvenv.cfg
+
```

Comment:
we can also include one more block which cleans up pycache and `*.pyc`
```bash
find ${{targets.contextdir}} -type d -name '__pycache__' -exec rm -rf {} +
find ${{targets.contextdir}} -type f -name '*.pyc' -exec rm -rf {} +
```

Reply:
wondering if we can also append the following given a lot of tests directory are showing up in package.  
```bash
find ${{targets.destdir}}/usr/share/${{package.name}}/.venv/lib/python${{vars.py-version}}/site-packages/ -type d -name "tests" -exec rm -rf {} +
```